### PR TITLE
Game Systems

### PR DESCRIPTION
**Pull request by SJuliez**
This is another step towards supporting other game types, oriented mostly towards SBF, but the AbstractGame and AbstractGameManager changes will benefit all. As before this PR should ideally not show any visible changes.
The code changes are:
- moved up deployment table handling from Game to AbstractGame and removed the focus on Entity
- added a Deployable interface for deployable units/objects. This is distinct from InGameObject as there are InGameObjects such as HexTarget that cant be deployed. It is possible that having Targetable extend InGameObject was not a wise move by me
- moved up the pendingActions list to AbstractGame
- moves the answer to "should this phase be skipped now" from GamePhase into the *Game classes as this is game and situation specific
- copied a whole lot of phase handling from GameManager to SBFGameManager. Most of it is commented out but I added a bit of working stuff. I try to work through the steps but this can't be done in one PR. SBFGameManager is already 1000 lines without doing much so it's a worthy companion to GameManager :)

I ran the tests, played a game with Princess and pulled these changes into MHQ and started a story arc fight. All seemed to work.

### CODE REVIEW

**Comment by HammerGS** [comment]
Now has conflicts.

File: megamek/src/megamek/common/Game.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public class Game extends AbstractGame implements Serializable, PlanetaryConditi
     private int victoryPlayerId = Player.PLAYER_NONE;
     private int victoryTeam = Player.TEAM_NONE;
 
-    private Hashtable<Integer, Vector<Entity>> deploymentTable = new Hashtable<>();
+//    private Hashtable<Integer, Vector<Entity>> deploymentTable = new Hashtable<>();
```
Comment (Sleet01) [inline megamek/src/megamek/common/Game.java:118]: Please clean this up.

**Review by Sleet01 — CHANGES_REQUESTED** [review]
One commented line, and a few codeQL notices that should be cleaned up (I believe the style guide calls out `@override` as being required).

Other than that, looks good - very exciting!

**Comment by SJuliez** [comment]
OK cleaned up. I believe the missing overrides were in the other PR, but I corrected them. Not seeing any warnings here.

**Review by Sleet01 — APPROVED** [review]

**Pull request merged**